gdal_subsetter/transform.py

```diff
@@ -856,28 +896,32 @@ def is_stackable(self, files_to_stack: List[str]) -> bool:
 
             formats.add(os.path.splitext(file_to_stack)[1])
 
-        same_projection = len(projections) == 1
-        same_geotransform = len(geotransforms) == 1
-        same_dtype = len(dtypes) == 1
-        same_xsize = len(xsizes) == 1
-        same_ysize = len(ysizes) == 1
+        same_grid = all(
+            len(grid_property_values) == 1
+            for grid_property_values
+            in (projections, geotransforms, dtypes, xsizes, ysizes)
+        )
```

Comment:
Hopefully this doesn't sacrifice readability in an attempt to be DRY.

The good news, though, is that there are unit tests for this function since a couple of PRs ago, so we know the behaviour remains consistent 🎉